lib/client.js

```diff
@@ -112,19 +112,26 @@ Client.prototype.send = function() {
   // Respect IRC line length limit of 512 chars
   // https://tools.ietf.org/html/rfc2812#section-2.3
   if (message.length > 512) {
-    var parts   = message.split(':');
-    var command = parts[1];
-    var payload = parts[2];
+    var payload_marker  = message.indexOf(' :');
+    var command = message.substr(0, payload_marker);
+    var payload = message.substr(payload_marker + 2);
 
     var chunks = payload.split(' ');
     var chunk = [];
     var _chunk;
     while (chunks.length) {
       chunk.push(chunks.shift());
-       _chunk = ':' + command + ':' + chunk.join(' ') + '\r\n';
+      _chunk = command + ' :' + chunk.join(' ');
       if (_chunk.length > 450 || !chunks.length) {
+
+        // handle long words with no spaces
+        if (_chunk.length > 510) {
```

Comment:
Why are we comparing the chunk to lengths of 450 and 510? I understand 510 is the max 512 minus CR + LF.